docs/spec/fastsync/fastsync.md

```diff
@@ -0,0 +1,1164 @@
+		 
+   - [Informal Problem statement](#Informal-Problem-statement): For
+   the 
+   general audience, that is, engineers who want to get an overview
+   over 
+   what the component is doing
+   from a bird's eye view. 
+
+   - [Sequential Problem statement](#Sequential-Problem-statement):
+     Provides a mathematical definition of the problem statement in
+     its sequential form, that is, ignoring the distributed aspect of
+     the implementation of the blockchain.
+
+  
+- [Part II](#part-ii---protocol-view): Here we introduce the protocol
+  view, that is, distributed aspects,
+  and temporal logic specifications. In this specification we provide a
+  description of Fastsync V2 (the protocol underlying the current Golang
+  implementation), and an analysis of Fastsync V2 that highlights 
+  several issues that
+  prevent achieving some of the desired fault-tolerance
+  properties.
+  
+    - [Environment/Assumptions/Incentives](#Environment/Assumptions/Incentives): 
+  distributed aspects, including timing and correctness assumptions.
+  
+    - [Distributed Problem Statement](#Distributed-Problem-Statement):
+	  We discuss [design choices](#Design-choices) that relax the
+	  termination requirements of the
+        (abstract) sequential problem statement to be solvable in a
+        distributed
+		unreliable environment, and we describe which messages should be
+	  exchanged. We also give [temporal
+	  properties](#Temporal-Properties) here that formalize safety and
+        liveness
+		properties
+		
+    - [Definitions](#Definitions): Describes inputs, outputs,
+      variables used by the protocol, auxiliary functions
+	  
+    - [Algorithm Invariants](#Algorithm-Invariants): invariants over
+      the protocol variables that the implementation should maintain.
+	  
+    - [FastSync V2](#FastSync-V2): gives an outline of the solution,
+      and details of the functions used (with preconditions,
+      postconditions, error conditions).
+	  
+    - [Analysis of Fastsync V2](#Analysis-of-Fastsync-V2): describes
+      undesireable scenarios of Fastsync V2, and why they violate
+      desireable temporal logic specification in an unreliable
+      distributed system.
+
+- [Part III](#part-iii---suggestions-for-an-improved-fastsync-implementation):
+  some suggestions on how to address the issues in the future.
+  
+   - [Desirable Temporal Properties](#Desirable-Temporal-Properties):
+     we give temporal properties that describe a fault-tolerant
+     version
+	 of FastSync.
+	 
+   - [Suggestions](#Suggestions)  to address the issues discussed in the analysis.
+  
+
+In this document we quite extensively use tags in order to be able to
+reference assumptions, invariants, etc. in future communication. In
+these tags we frequently use the following short forms:
+
+- TMBC: Tendermint blockchain
+- SEQ: for sequential specifications
+- FS: Fastsync
+- LIVE: liveness
+- INV: invariant
+- TERM: termination
```

Comment:
What is difference between liveness and termination? I would expect that it is the same and that one abbreviation is sufficient. On the other side we don't have abbreviation for safety properties.

Reply:
As you noted below, TERM was mostly addressing safety properties upon termination. I will remove TERM here, as we have changed the tags